core/Scheduler/Timetable.php

```diff
@@ -149,4 +163,71 @@ public function readFromOption()
 
         $this->timetable = $unserializedTimetable === false ? array() : $unserializedTimetable;
     }
+
+    /**
+     * Read the retry list option from the database
+     *
+     * @throws \Throwable
+     */
+    private function readRetryList()
+    {
+        Option::clearCachedOption(self::RETRY_OPTION_STRING);
+        $retryData = Option::get(self::RETRY_OPTION_STRING);
+        $unserializedRetryList = Common::safe_unserialize($retryData);
+
+        $this->retryList = $unserializedRetryList === false ? array() : $unserializedRetryList;
+    }
+
+    /**
+     * Save the retry list option to the database
+     */
+    public function saveRetryList()
+    {
+        Option::set(self::RETRY_OPTION_STRING, serialize($this->retryList));
+    }
+
+    /**
+     * Remove a task from the retry list
+     *
+     * @param string $taskName
+     */
+    public function clearRetryCount(string $taskName)
+    {
+        if (isset($this->retryList[$taskName])) {
+            unset($this->retryList[$taskName]);
+            $this->saveRetryList();
+        }
+    }
+
+    /**
+     * Increment the retry counter for a task
+     *
+     * @param string $taskName
+     */
+    public function incrementRetryCount(string $taskName)
+    {
+        if (!isset($this->retryList[$taskName])) {
+            $this->retryList[$taskName] = 0;
+        }
+        $this->retryList[$taskName]++;
+        $this->saveRetryList();
+    }
+
+    /**
+     * Return the current number of retries for a task
+     *
+     * @param string $taskName
+     *
+     * @return int
+     */
+    public function getRetryCount(string $taskName) : int
+    {
+        // Ignore excessive retry counts, workaround for SchedulerTest mock
+        if (!isset($this->retryList[$taskName]) || $this->retryList[$taskName] > 10000) {
```

Comment:
would we need to here also first call again `readRetryList` in case a difference process updated it ?